### PR TITLE
Rewrite BitbucketPullRequestPoller

### PR DESCRIPTION
**Pull request by chapipo**
- Use httpclient
- Support Oauth2 authentification

I removed some arguments of the poller (project, useTimestamp) because I was not sure about how they have to work. If needed I will work to reintroduce them.

## Contributor Checklist:

* [X] I have updated the unit tests
* [ ] I have created a file in the `master/buildbot/newsfragments` directory (and read the `README.txt` in that directory)
* [X] I have updated the appropriate documentation

### CODE REVIEW

**Comment by p12tic** [comment]
Could we split the changes into separate commits so that it's possible to review it? E.g. the httpclient rewrite could go first and potentially as a separate PR.

**Comment by chapipo** [comment]
Will try to do this but the httpclient rewrite is the biggest :)

File: master/buildbot/changes/bitbucket.py [hunk 1]
```diff
@@ -14,184 +14,282 @@
...
-                 pollInterval=10 * 60,
-                 useTimestamps=True,
-                 category=None,
-                 project='',
-                 pullrequest_filter=True,
-                 encoding=_UNSPECIFIED,
-                 pollAtLaunch=False
-                 ):
+    def __init__(self, owner, slug, **kwargs):
+        name = kwargs.get("name")
```
Comment (sa2ajj) [inline master/buildbot/changes/bitbucket.py]: two comments about these lines:
* could we use `kwargs.setdefault("name", ...)` instead?
* could you please replace string concatenations with a string interpolation (`"xxx {} xxx {} xxx`.format(...)`)? interpolations are easier to read and easier to review when changes are made

File: master/buildbot/changes/bitbucket.py [hunk 2]
```diff
@@ -14,184 +14,282 @@
...
+                    pullrequest_filter=True,
+                    codebase='',
+                    oauth_key=None,
+                    oauth_secret=None,
+                    repository_type="https",
+                    debug=False,
+                    **kwargs):
+        if repository_type not in ["https", "ssh"]:
+            config.error(
+                "repository_type must be one of {https, svn, git, ssh}")
```
Comment (sa2ajj) [inline master/buildbot/changes/bitbucket.py]: i am not sure how the error message corresponds to the condition... please add a comment clarifying the relationship or change the condition and/or the message

File: master/buildbot/changes/bitbucket.py [hunk 3]
```diff
@@ -14,184 +14,282 @@
...
+                        codebase='',
+                        oauth_key=None,
+                        oauth_secret=None,
+                        debug=False,
+                        repository_type="https",
+                        **kwargs):
+
+        yield super().reconfigService(name=self.name, **kwargs)
+
+        http_headers = {'User-Agent': 'Buildbot'}
```
Comment (sa2ajj) [inline master/buildbot/changes/bitbucket.py]: are you sure that the set of headers is not going to increase?  if not, please re-format as
```
http_headers = {
    'User-Agent': 'Buildbot'.
}
```

**Pull request closed without merging**